src/test/java/uk/gov/pay/api/it/CreatePaymentIT.java

```diff
@@ -202,6 +202,53 @@ public void createAChargeWithMetadataAgreementNotFound() throws IOException {
 
         connectorMockClient.verifyCreateChargeConnectorRequest(GATEWAY_ACCOUNT_ID, createChargeRequestParams);
     }
+    
+    @Test
+    public void createPayment_responseWith400_whenIncorrectAuthorisationModeForSavePaymentInstrumentToAgreement() {
+        publicAuthMockClient.mapBearerTokenToAccountId(API_KEY, GATEWAY_ACCOUNT_ID);
+
+        CreateChargeRequestParams createChargeRequestParams = aCreateChargeRequestParams()
+                .withAmount(100)
+                .withDescription(DESCRIPTION)
+                .withReference(REFERENCE)
+                .withAuthorisationMode(AuthorisationMode.AGREEMENT)
+                .withSetUpAgreement(VALID_AGREEMENT_ID)
+                .build();
+        connectorMockClient.respondIncorrectAuthorisationModeForSavePaymentInstrumentToAgreement_whenCreateCharge(GATEWAY_ACCOUNT_ID, VALID_AGREEMENT_ID, "error message from connector");
+
+        postPaymentResponse(paymentPayload(createChargeRequestParams))
+                .statusCode(400)
+                .contentType(JSON)
+                .body("field", is(nullValue()))
+                .body("code", is("P0104"))
```

Comment:
Should this be expecting `P0102`? I think this code and it's assertion don't line up with the commit description

Reply:
No, it’s P0104. Mistake in the commit message and PR description (just fixed).